web/src/test/java/org/springframework/security/web/authentication/www/BasicAuthenticationFilterTests.java

```diff
@@ -81,6 +82,15 @@ public void setUp() {
 		given(this.manager.authenticate(rodRequest)).willReturn(rod);
 		given(this.manager.authenticate(not(eq(rodRequest)))).willThrow(new BadCredentialsException(""));
 		this.filter = new BasicAuthenticationFilter(this.manager, new BasicAuthenticationEntryPoint());
+		this.filter.setAuthenticationConverter(new BasicAuthenticationConverter() {
+			@Override
+			public UsernamePasswordAuthenticationToken convert(final HttpServletRequest request) {
+				if (request.getServletPath().equalsIgnoreCase("/ignored-request")) {
+					return null;
+				}
+				return super.convert(request);
+			}
+		});
```

Comment:
You can set the `AuthenticationConverter` into the filter in a specific test, there is no need to add it for every test method in `@BeforeEach`.

Reply:
In addition to that, we recommend using a delegate than extend the existing `BasicAuthenticationConverter`, something like:

```java
static class MyAuthenticationConverter implements AuthenticationConverter {

    private final BasicAuthenticationConverter delegate = new BasicAuthenticationConverter();

    @Override
	public UsernamePasswordAuthenticationToken convert(final HttpServletRequest request) {
		// check if request matches, maybe use RequestMatcher here
		return this.delegate.convert(request);
	}

}
```

Sometimes people refer to the tests as implementation examples, so it is better to align with what the team recommends and uses.

Reply:
A delegate is not possible here because the setter is:

```java
public void setAuthenticationConverter(BasicAuthenticationConverter authenticationConverter) {}
```

It's not possible to switch to:

```java
public void setAuthenticationConverter(AuthenticationConverter authenticationConverter) {}
```

...and change the field type to `AuthenticationConverter` because that breaks other things in the filter.

How would you like to proceed?